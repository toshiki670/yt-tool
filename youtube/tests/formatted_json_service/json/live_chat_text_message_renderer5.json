{
  "replayChatItemAction": {
    "actions": [
      {
        "addChatItemAction": {
          "item": {
            "liveChatTextMessageRenderer": {
              "message": {
                "runs": [
                  {
                    "emoji": {
                      "emojiId": "👨‍❤‍👨",
                      "shortcuts": [":couple_with_heart_man_man:"],
                      "searchTerms": ["couple", "with", "heart", "man"],
                      "image": {
                        "thumbnails": [
                          {
                            "url": "https://fonts.gstatic.com/s/e/notoemoji/15.1/1f468_200d_2764_200d_1f468/72.png"
                          }
                        ],
                        "accessibility": {
                          "accessibilityData": { "label": "👨‍❤‍👨" }
                        }
                      },
                      "variantIds": [
                        "👨‍❤‍👨",
                        "👨🏻‍❤‍👨🏻",
                        "👨🏻‍❤‍👨🏼",
                        "👨🏻‍❤‍👨🏽",
                        "👨🏻‍❤‍👨🏾",
                        "👨🏻‍❤‍👨🏿",
                        "👨🏼‍❤‍👨🏻",
                        "👨🏼‍❤‍👨🏼",
                        "👨🏼‍❤‍👨🏽",
                        "👨🏼‍❤‍👨🏾",
                        "👨🏼‍❤‍👨🏿",
                        "👨🏽‍❤‍👨🏻",
                        "👨🏽‍❤‍👨🏼",
                        "👨🏽‍❤‍👨🏽",
                        "👨🏽‍❤‍👨🏾",
                        "👨🏽‍❤‍👨🏿",
                        "👨🏾‍❤‍👨🏻",
                        "👨🏾‍❤‍👨🏼",
                        "👨🏾‍❤‍👨🏽",
                        "👨🏾‍❤‍👨🏾",
                        "👨🏾‍❤‍👨🏿",
                        "👨🏿‍❤‍👨🏻",
                        "👨🏿‍❤‍👨🏼",
                        "👨🏿‍❤‍👨🏽",
                        "👨🏿‍❤‍👨🏾",
                        "👨🏿‍❤‍👨🏿"
                      ],
                      "multiSelectorThumbnailRow": [
                        {
                          "thumbnails": [
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-light_shadow/v1/128px.svg"
                                }
                              ]
                            },
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-medium-light_shadow/v1/128px.svg"
                                }
                              ]
                            },
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-medium_shadow/v1/128px.svg"
                                }
                              ]
                            },
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-medium-dark_shadow/v1/128px.svg"
                                }
                              ]
                            },
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-dark_shadow/v1/128px.svg"
                                }
                              ]
                            }
                          ]
                        },
                        {
                          "thumbnails": [
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-shadow_light/v1/128px.svg"
                                }
                              ]
                            },
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-shadow_medium-light/v1/128px.svg"
                                }
                              ]
                            },
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-shadow_medium/v1/128px.svg"
                                }
                              ]
                            },
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-shadow_medium-dark/v1/128px.svg"
                                }
                              ]
                            },
                            {
                              "thumbnails": [
                                {
                                  "url": "https://fonts.gstatic.com/s/i/googleemojiextra/couple_heart-men-shadow_dark/v1/128px.svg"
                                }
                              ]
                            }
                          ]
                        }
                      ]
                    }
                  }
                ]
              },
              "authorName": { "simpleText": "投稿者名" },
              "authorPhoto": {
                "thumbnails": [
                  {
                    "url": "https://yt4.ggpht.com/ytc/AIdro_nYCq3vQPX0U2Noc90RInQ6Mrgm1hrsCmjBH380dv5hQ0U=s32-c-k-c0x00ffffff-no-rj",
                    "width": 32,
                    "height": 32
                  },
                  {
                    "url": "https://yt4.ggpht.com/ytc/AIdro_nYCq3vQPX0U2Noc90RInQ6Mrgm1hrsCmjBH380dv5hQ0U=s64-c-k-c0x00ffffff-no-rj",
                    "width": 64,
                    "height": 64
                  }
                ]
              },
              "contextMenuEndpoint": {
                "commandMetadata": {
                  "webCommandMetadata": { "ignoreNavigation": true }
                },
                "liveChatItemContextMenuEndpoint": {
                  "params": "Q2g0S0hBb2FRMUJYV21sTFRIUXhTVzlFUmxGZlEzZG5VV1J4UVdOSExWRWFLU29uQ2hoVlEyOUJSRWRQUTBoME1HSklOVXN0VXpobFRGOXJkMmNTQzE4eGMwTTRXVU5ZWTNScklBRW9CRElhQ2hoVlEwRmhiamhqWVdKUmNYYzNRV3hvWXpOMlgyRk5XSGM0QWtnQVVBRSUzRA=="
                }
              },
              "id": "ChwKGkNQV1ppS0x0MUlvREZRX0N3Z1FkcUFjRy1R",
              "timestampUsec": "1735745874301790",
              "authorExternalChannelId": "UCAan8cabQqw7Alhc3v_aMXw",
              "contextMenuAccessibility": {
                "accessibilityData": { "label": "Chat actions" }
              }
            }
          },
          "clientId": "CPWZiKLt1IoDFQ_CwgQdqAcG-Q"
        }
      }
    ]
  },
  "videoOffsetTimeMsec": "3434106",
  "isLive": true
}
