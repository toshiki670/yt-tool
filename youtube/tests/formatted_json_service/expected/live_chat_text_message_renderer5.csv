id,authorExternalChannelId,postedAt,category,authorName,content
ChwKGkNQV1ppS0x0MUlvREZRX0N3Z1FkcUFjRy1R,UCAan8cabQqw7Alhc3v_aMXw,2025-01-02T00:37:54.301790+09:00,Text message,投稿者名,message: '👨‍❤‍👨'
